### PR TITLE
dev

### PR DESCRIPTION
**Pull request by Amitamaru**

### CODE REVIEW

File: src/main/java/com/marzhiievskyi/home_notes/dao/implementation/SearchDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+@RequiredArgsConstructor
+public class SearchDaoImpl extends JdbcDaoSupport implements SearchDao {
+
+    private final DataSource dataSource;
+
+    private final JdbcTemplate jdbcTemplate;
+
+    @PostConstruct
+    private void init() {
+        setDataSource(dataSource);
```
Comment (demologin) [inline src/main/java/com/marzhiievskyi/home_notes/dao/implementation/SearchDaoImpl.java:31]: ????

File: src/main/java/com/marzhiievskyi/home_notes/service/SearchService.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    private final ValidationUtils validationUtils;
+
+    public List<TagResponseDto> getTagsByNoteId(Long noteId) {
+        return searchDao.getTagsByNoteId(noteId);
+    }
+
+    public ResponseEntity<Response> findTagsByPart(SearchTagsRequestDto searchTagRequest, String accessToken) {
+
+        validationUtils.validationRequest(searchTagRequest);
+        userDao.getUserIdByAccessToken(accessToken);
```
Comment (demologin) [inline src/main/java/com/marzhiievskyi/home_notes/service/SearchService.java]: ????

File: src/main/java/com/marzhiievskyi/home_notes/dao/implementation/SearchDaoImpl.java (hunk 1 above)
Comment (Amitamaru) [reply src/main/java/com/marzhiievskyi/home_notes/dao/implementation/SearchDaoImpl.java:31]: from doc JdbcDaoSupport
Set the JDBC DataSource to be used by this DAO.
Create a JdbcTemplate for the given DataSource. Only invoked if populating the DAO with a DataSource reference!
Can be overridden in subclasses to provide a JdbcTemplate instance with different configuration, or a custom JdbcTemplate subclass.

File: src/main/java/com/marzhiievskyi/home_notes/service/SearchService.java (hunk 2 above)
Comment (Amitamaru) [reply src/main/java/com/marzhiievskyi/home_notes/service/SearchService.java]: refactor to "findUserIdIByTokenOrThrowException" name method =)

**Pull request merged**